### PR TITLE
week3 homework

### PR DESCRIPTION
**Pull request by Dumie1**

### CODE REVIEW

File: homework-submission/myapp/src/Components/Data.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  addTodo = event => {
+    event.preventDefault();
+
+    const descriptionInput = this.descriptionInput.current.value;
+    const deadlineInput = this.deadlineInput.current.value;
+    const newState = this.state;
+
+    const Todos = newState.Todos;
+    Todos.push({
+      id: Todos.length + 1,
```
Comment (Jawhar-B) [inline homework-submission/myapp/src/Components/Data.js]: What is the `id` for the next one of this if this `todo` has been deleted?

Well in your case you are re-rendering the whole thing, which is fine.

But in a different situation you will face a Bug may be hard to follow or fix, so may have to re-think about how do I specify an `Id` for a new Item.

File: homework-submission/myapp/src/Components/Data.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    this.setState(newState);
+
+    this.descriptionInput.current.value = '';
+    this.deadlineInput.current.value = '';
+  };
+
+  deleteTodo = id => {
+    const arr = this.state.Todos;
+    arr.splice(id, 1);
```
Comment (Jawhar-B) [inline homework-submission/myapp/src/Components/Data.js]: same as a previous comment, this is mutating the state immediately, perhaps `.filter` Method ?

Remember the checking test:
```javascript
    console.log(this.state.Todos)
    arr.splice(id, 1);
    console.log(this.state.Todos)
```

File: homework-submission/myapp/src/Components/Data.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    if (this.state.Todos.length === 0)
+      return <p className='noItems'>No items...</p>;
+    const textStyle = {
+      textDecoration: 'line-through',
+      backgroundColor: '#0455d8'
+    };
+
+    return (
+        <ul>
+        {Todos.map((todo, id) => {
```
Comment (Jawhar-B) [inline homework-submission/myapp/src/Components/Data.js]: where `Todos` come from? 🙃

Comment (Dumie1) [reply homework-submission/myapp/src/Components/Data.js]: Instead of `Todos.map` , my corrections are as follows `this.state.Todos.map` ... i hope im on the right track

File: homework-submission/myapp/src/Components/Data.js (hunk 2 above)
Comment (Dumie1) [reply homework-submission/myapp/src/Components/Data.js]: Hahaha i get what you mean. Probably `.splice()` method was easy for me to use. However thanks @Jawhar-B for the constructive feedback. i will try using `.filter()` method.
